app/js/lib/foxbox/foxbox.js

```diff
@@ -389,6 +389,71 @@ export default class Foxbox extends Service {
     return this[p.db].setTag.apply(this[p.db], arguments);
   }
 
+  subscribeToNotifications(resubscribe = false) {
+    const settings = this[p.settings];
+    const boxHost = this[p.net].origin;
+    if (!navigator.serviceWorker) {
+      return Promise.reject('No service worker supported');
+    }
+
+    if (settings.pushEndpoint && settings.pushPubKey && !resubscribe) {
+      return Promise.resolve();
+    }
+
+    return navigator.serviceWorker.ready.then((reg) => {
+      reg.pushManager.subscribe({userVisibleOnly: true})
+      .then((subscription) => {
+        const endpoint = subscription.endpoint;
+        const key = subscription.getKey ? subscription.getKey('p256dh') : '';
+        settings.pushEndpoint = endpoint;
+        settings.pushPubKey = btoa(String.fromCharCode.apply(null,
+          new Uint8Array(key)));
+
+        // Send push information to the server
+        // XXX: We will need some library to write taxonomy messages
+        const pushConfigurationMsg = [[
+            [{
+              id: 'setter:subscribe.webpush@link.mozilla.org'
+            }], {
+              Json: {
+                subscriptions: [{
+                  public_key: settings.pushPubKey,
+                  push_uri: settings.pushEndpoint
```

Comment:
Do we really need to store `pushPubKey` and `pushEndpoint` in the settings? It looks like they are always set before being used in the push configuration.
If that's the case, maybe we just need something like `settings.pushConfigured` that defaults to `false`.